rich/_wrap.py

```diff
@@ -17,40 +20,71 @@ def words(text: str) -> Iterable[Tuple[int, int, str]]:
         word_match = re_word.match(text, end)
 
 
-def divide_line(text: str, width: int, fold: bool = True) -> List[int]:
-    divides: List[int] = []
-    append = divides.append
-    line_position = 0
+def divide_line(text: str, width: int, fold: bool = True) -> list[int]:
```

Comment:
I'm assuming `cell_offset` is keeping track of how much space is already occupied in the line we're currently filling, right?
I _personally_ don't love the name but maybe that could be mitigated with a short comment next to it?